compressed_image_transport/package.xml

```diff
@@ -17,10 +17,12 @@
   <build_depend>cv_bridge</build_depend>
   <build_depend>dynamic_reconfigure</build_depend>
   <build_depend>image_transport</build_depend>
+  <build_depend>libturbojpeg</build_depend>
 
   <run_depend>cv_bridge</run_depend>
   <run_depend>dynamic_reconfigure</run_depend>
   <run_depend>image_transport</run_depend>
+  <run_depend>libturbojpeg</run_depend>
```

Comment:
I suggest switchin to package format 2 and declaring this only as `<exec_depend>` - otherwise all dependent packages would need to link to turbojpeg, which is not needed.

Reply:
Good point, will do so.

Reply:
Actually, at the moment we `#include <turbojpeg.h>` in `compressed_subscriber.h`, which makes it a build export dependency, right?

We could maybe replace the `tjhandle` with a `void*` and avoid the include in the header. Or have a pimpl-style `std::unique_ptr<Private>` member, that way we can define the `tjhandle` in the cpp file.

Reply:
I think the reasoning here is the same as with the ABI compatibility - technically, it would add a build-export dependency if someone linked to the plugin, but doing so is highly improbable...

Reply:
Ok!